src/main/java/org/jabref/gui/openoffice/OpenOfficePanel.java

```diff
@@ -365,41 +368,53 @@ private void connectManually() {
     }
 
     private void connect() {
-        DialogPane dialogPane = new DialogPane();
-        ProgressIndicator indicator = new ProgressIndicator(ProgressIndicator.INDETERMINATE_PROGRESS);
-        dialogPane.setContent(indicator);
-        FXDialog progressDialog = dialogService.showCustomDialog(Localization.lang("Autodetecting paths..."), dialogPane, ButtonType.CANCEL);
-        progressDialog.show();
+
+        // dialogService.showProgressDialogAndWait(Localization.lang("Autodetecting paths..."), , ButtonType.CANCEL);
 
         try {
-            ooPrefs = jabRefPreferences.getOpenOfficePreferences();
-            // Add OO JARs to the classpath
-            loadOpenOfficeJars(Paths.get(ooPrefs.getInstallationPath()));
 
-            BackgroundTask
-                          .wrap(this::createBibBase)
-                          .onFinished(() -> progressDialog.close())
-                          .onSuccess(ooBase -> {
-                              this.ooBase = ooBase;
+            Task<OOBibBase> task = new Task<OOBibBase>() {
 
-                              if (ooBase.isConnectedToDocument()) {
-                                  dialogService.notify(Localization.lang("Connected to document") + ": " + ooBase.getCurrentDocumentTitle().orElse(""));
-                              }
+                @Override
+                protected OOBibBase call() throws Exception {
 
-                              // Enable actions that depend on Connect:
-                              selectDocument.setDisable(false);
-                              pushEntries.setDisable(false);
-                              pushEntriesInt.setDisable(false);
-                              pushEntriesEmpty.setDisable(false);
-                              pushEntriesAdvanced.setDisable(false);
-                              update.setDisable(false);
-                              merge.setDisable(false);
-                              manageCitations.setDisable(false);
-                              exportCitations.setDisable(false);
+                    updateProgress(ProgressBar.INDETERMINATE_PROGRESS, ProgressBar.INDETERMINATE_PROGRESS);
```

Comment:
I found that I also could use the normal progressbar dialog with indeterminate progress.
However, the main problem is now that I need to run the createBibBase in a background task and not on the FX task which is now the case.
Is there a way to update the progress on our BackgroundTaskWrapper? I can't really get my head around those executor stuff.